### PR TITLE
Equalize mouselook sensitivity

### PR DESCRIPTION
**Pull request by LidMop**
This PR equalizes horizontal and vertical mouselook sensitivity while preserving existing sensitivity preferences. Users no longer need to manually edit the prefs file and can just use the slider in the Controls dialog.

The root cause is `mouse_idle()` and `mask_in_absolute_positioning_information()` unwisely using different (and unusual) units for yaw and pitch, but forgetting that detail when it scales the mouse inputs by sensitivity. I want to refactor those functions to use a common angular unit but the changes would conflict with #117, so I'm holding off for now.

### CODE REVIEW

**Comment by Hopper262** [comment]
Interesting! Bungie had plenty of opportunity to fix this, but it wouldn't be the strangest oversight I guess. Keyboard aiming uses the same acceleration for both directions, so it makes sense to do the same for mouse.

I do wonder if it'd be better to keep mouse (and joystick) ignorant of the asymmetry, and scale inside `mask_in_absolute_positioning_information()` instead. Maybe you're already thinking along those lines.

**Comment by LidMop** [comment]
> I do wonder if it'd be better to keep mouse (and joystick) ignorant of the asymmetry, and scale inside `mask_in_absolute_positioning_information()` instead. Maybe you're already thinking along those lines.

To clarify, I want "dx units" and "dy units" completely eliminated in favor of Bungie angular units, with symmetric scaling everywhere. The current asymmetry is solely due to using one unit ("dx units") for yaw and another unit ("dy units") for pitch. The 4x factor I added in this PR is really a conversion factor to convert from scaled mouse input to "dy units", while the conversion factor to "dx units" is 1. (I probably should have expressed the code in that manner, but it'll get reworked soon anyway.)

**Comment by Hopper262** [comment]
This might be old news to you, but I didn't realize it before: `mask_in_absolute_positioning_information()` thinks the encoded yaw and pitch fields within input_flags have the same range but different precision, while `physics_update()` thinks the fields have the same precision but different ranges. If the encoding and decoding were in sync, the current mouse code would be fine.

**Comment by LidMop** [comment]
I mentally split `mask_in_absolute_positioning_information()` into two steps: 1) converting its yaw/pitch parameters to Bungie angular units, and 2) encoding those Bungie angles in action flags. In this mental model, the encoding step and the decoding step done in `physics_update()` are consistent, but the conversion step scales the yaw and pitch parameters differently.

This asymmetric scaling was intentional, according to a Bungie comment above the function:
```
/* deltas of zero are ignored; all deltas must be in [-FIXED_ONE,FIXED_ONE] which will be scaled
	to the maximum for that value */
```

(Note that the comment is wrong, as are the names of the constants, `MAXIMUM_ABSOLUTE_YAW` and `MAXIMUM_ABSOLUTE_PITCH`: the actual maximums are about half as large.)

In other words, Bungie defined the parameters as **unitless coefficients** and punted the question of units to calling code (how rude!). I had forgotten about that until just now. Back when I did #52, I conceptually redefined these coefficients as angular quantities with equivalently-scaled units, "dx units" and "dy units", in an attempt to get a more meaningful sense of yaw/pitch deltas in `mouse_idle()` without extra refactoring.

**Comment by Hopper262** [comment]
Ah, thank you for explaining! My mental model came out differently than yours. In my reading, the original intent was:

1. `mouse_idle` and `test_mouse` translate and scale mouse velocities into a [-FIXED_ONE, FIXED_ONE] range, with the (hardcoded, later user-adjustable) sensitivity defining how many mouse units it takes to hit the cap
2. `mask_in_absolute_positioning_information` compresses the [-FIXED_ONE, FIXED_ONE] range into the limited resolution of the input_flags, keeping as much precision as fits
3. `physics_update` decompresses the input flags, and decides how to apply the reconstructed [-FIXED_ONE, FIXED_ONE] range into Bungie angles, possibly by scaling based on `maximum_angular_velocity`

Obviously `physics_update` doesn't follow that plan in the shipped code. I assume at some point, Bungie realized that preserving the low bits (for slow movement) was more important than the upper limit, and changed `physics_update` but forgot to update `mask` to maintain a constant precision instead of range.

I never saw `mask` in its current form as knowing or caring about actual Bungie angles, so that's why we've been thinking about it differently. Your view -- that `mask` explicitly computes angles for storage -- is valid in light of how `physics_update` works, and it's certainly easier to leave `physics_update` as is, so for your refactor we both want the same end result.

I agree we should address the unitless dodge too. Maybe `mask` should continue to take _fixed values, but with FIXED_ONE equivalent to 1 Bungie angle? That way, `mask` can still accept high-res velocities from mouse and joystick input, to better share code for smoothing or future input_flags expansion, but the caller understands the end physics result.

How about I accept this and #117 into a branch, and you can refactor on top of that? When we're all happy with the structure, it'll get merged back to master (definitely before 1.3!).

**Comment by LidMop** [comment]
The design evolution you describe sounds very plausible!

> Maybe `mask` should continue to take _fixed values, but with FIXED_ONE equivalent to 1 Bungie angle? That way, `mask` can still accept high-res velocities from mouse and joystick input, to better share code for smoothing or future input_flags expansion, but the caller understands the end physics result.

Agreed!

> How about I accept this and #117 into a branch, and you can refactor on top of that? When we're all happy with the structure, it'll get merged back to master (definitely before 1.3!).

I'd be happy to!

**Comment by Hopper262** [comment]
Awesome, see you in the next PR! :)
